src/ray/raylet/scheduling/cluster_resource_data.cc

```diff
@@ -408,6 +415,13 @@ std::string NodeResources::DictString() const {
   return buffer.str();
 }
 
+bool NodeResources::HasGPU() const {
+  if (predefined_resources.size() <= GPU) {
```

Comment:
should we use std::array for `predefined_resources`? It seems easy to make mistake and we know the number of predefined resources.

Reply:
yeah @raulchen is working on refactoring the resource representation. https://github.com/ray-project/ray/pull/22854